shared/validate.js

```diff
@@ -50,32 +50,38 @@ module.exports = {
     return BbPromise.resolve(currentErrors.concat(namespaceErrors));
   },
 
-  validateFunctions(errors) {
+  validateApplications(errors) {
+    let functionNames = [];
+    let containerNames = [];
+
     const currentErrors = Array.isArray(errors) ? errors : [];
-    const functions = this.serverless.service.functions;
-    const functionNames = Object.keys(functions);
     let functionErrors = [];
 
-    if (!functionNames.length) {
-      functionErrors.push('You must define at least one function to deploy under the functions: key.');
+    const functions = this.serverless.service.functions;
+    if (functions) {
```

Comment:
you should check for length of array as well, if I define the function key you'll get into this condition.
```javascript
if (functions && Array.isArray(functions) && functions.length > 0)
```

Reply:
`functions` is an object. Now, I check if it is not empty